tests/migrations/test_optimizer.py

```diff
@@ -270,6 +270,35 @@ def test_optimize_through_create(self):
             app_label="testapp",
         )
 
+        # This could be optimized a bit more but at least it generates a valid
+        # set of operations.
+        self.assertOptimizesTo(
+            [
+                migrations.CreateModel('Book', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Person', [('name', models.CharField(max_length=255))]),
+                migrations.AddField('book', 'author', models.ForeignKey('test_app.Person', models.CASCADE)),
+                migrations.CreateModel('Review', [('book', models.ForeignKey('test_app.Book', models.CASCADE))]),
+                migrations.CreateModel('Reviewer', [('name', models.CharField(max_length=255))]),
+                migrations.AddField('review', 'reviewer', models.ForeignKey('test_app.Reviewer', models.CASCADE)),
+                migrations.RemoveField('book', 'author'),
+                migrations.DeleteModel('Person'),
+            ],
+            [
+                migrations.CreateModel('Person', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Book', [
+                    ('name', models.CharField(max_length=255)),
+                    ('author', models.ForeignKey('test_app.Person', models.CASCADE)),
+                ]),
+                migrations.CreateModel('Reviewer', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Review', [
+                    ('book', models.ForeignKey('test_app.Book', models.CASCADE)),
+                    ('reviewer', models.ForeignKey('test_app.Reviewer', models.CASCADE)),
+                ]),
+                migrations.RemoveField('book', 'author'),
```

Comment:
This could be optimized even more if `RemoveField` had any sort of context about which kind of field it is removing as we can't assume it isn't a field pointing at one of the model created in between this operation and the `CreateModel('Book')`.